framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -541,7 +517,14 @@ public Boolean visitDeclared_Typevar(AnnotatedDeclaredType subtype, AnnotatedTyp
 
     @Override
     public Boolean visitDeclared_Union(AnnotatedDeclaredType subtype, AnnotatedUnionType supertype, VisitHistory visited) {
-        return visitUnionSupertype(subtype, supertype, visited);
+        Types types = checker.getTypeUtils();
+        for (AnnotatedDeclaredType supertypeAltern : supertype.getAlternatives()) {
+            if (TypesUtils.isErasedSubtype(types, subtype.getUnderlyingType(), supertypeAltern.getUnderlyingType())
```

Comment:
For Intersection_Declared it would be equivalent.  It's not for  [Intersection_Intersection](https://github.com/typetools/checker-framework/pull/772/files/54a1731810bb71ca587bb9e25a87498e539e6421#diff-ace109c2136c5d7b905d5679ee384f43R567) nor [Intersection_Primitive](https://github.com/typetools/checker-framework/pull/772/files/54a1731810bb71ca587bb9e25a87498e539e6421#diff-ace109c2136c5d7b905d5679ee384f43R553).